### ISSUE TITLE
Update Project Dependencies

### ISSUE THREAD

**Issue by Impure-King** [Impure-King/base-sentinex#6]
# Description

Update the ``setup.py`` file and resolve any dependency errors by properly handling the versioning. Additionally, add an examples_require in the setup to accomodate for issue #5.

# Acceptance Criteria

- [x] Add jax and jaxlib to install_requires in setup.py.
- [x] Add a new examples section to extras_require in setup.py.
- [x] Add tensorflow-datasets to the new examples extra (for loading MNIST).

**Comment by Impure-King** [comment]
Adding bug label:
* ``jax.tree_map``: Updated and removed from main API.
